### PR TITLE
bubble sort content added

### PR DESCRIPTION
**Pull request by shonali2600**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#206

### CODE REVIEW

**Comment by shonali2600** [comment]
@HarshCasper Kindly review my PR and let me know if any changes to be made.

**Comment by shonali2600** [comment]
@HarshCasper Kindly review my PR

File: docs/sorting-algorithms/bubble-sort.md [hunk 1]
```diff
@@ -3,4 +3,162 @@ title: Bubble Sort
...
+### Output:
+
+```
+1 2 3 5 6 8 9 10
+```
+
+## Complexity Analysis
+
+- Time  complexity - O(n^2)
+- Space complexity - O(1)
```
Comment (HarshCasper) [inline docs/sorting-algorithms/bubble-sort.md:158]: Wrap them around backticks

File: docs/sorting-algorithms/bubble-sort.md [hunk 2]
```diff
@@ -3,4 +3,162 @@ title: Bubble Sort
...
+        print(i, end=" ")
+
+```
+
+</TabItem>
+</Tabs>
+
+## Sample Input and Output
+
+### Input:
```
Comment (HarshCasper) [inline docs/sorting-algorithms/bubble-sort.md:143]: Use Heading-4

File: docs/sorting-algorithms/bubble-sort.md [hunk 3]
```diff
@@ -3,4 +3,162 @@ title: Bubble Sort
...
+
+```
+2 5 1 6 3 9 10 8
+```
+
+### Output:
```
Comment (HarshCasper) [inline docs/sorting-algorithms/bubble-sort.md:149]: Use Heading-4

File: docs/sorting-algorithms/bubble-sort.md [hunk 4]
```diff
@@ -3,4 +3,162 @@ title: Bubble Sort
...
+		//For loop for printing out the array after sorting
+		cout << arr[i] << " ";
+	}
+}
+void bubblesort(int arr[], int n)
+{
+	int temp;
+	for (int i = 0; i < n - 1; i++)
+	{
+		bool sorted = true;	// declaring a sorted variable which will check if array is already sorted or not
```
Comment (HarshCasper) [inline docs/sorting-algorithms/bubble-sort.md:90]: Don't use comments beside the line of code. Use them above it.

File: docs/sorting-algorithms/bubble-sort.md [hunk 5]
```diff
@@ -3,4 +3,162 @@ title: Bubble Sort
...
+		for (int j = 0; j < n - 1 - i; j++)
+		{
+			//Inner for loop
+			if (arr[j] > arr[j + 1])
+			{
+				temp = arr[j + 1];	//Swapping the elements, temp is used to store the temporary variable
```
Comment (HarshCasper) [inline docs/sorting-algorithms/bubble-sort.md:96]: Don't use comments beside the line of code. Use them above it.

File: docs/sorting-algorithms/bubble-sort.md [hunk 6]
```diff
@@ -3,4 +3,162 @@ title: Bubble Sort
...
+				arr[j + 1] = arr[j];
+				arr[j] = temp;
+				sorted = false;	// sorted value will be updated to false if there is swapping of elements occured
```
Comment (HarshCasper) [inline docs/sorting-algorithms/bubble-sort.md:99]: Don't use comments beside the line of code. Use them above it.

File: docs/sorting-algorithms/bubble-sort.md [hunk 7]
```diff
@@ -3,4 +3,162 @@ title: Bubble Sort
...
+```cpp
+#include <bits/stdc++.h>
+using namespace std;	//For cin and cout
+void bubblesort(int[], int);
+int main()
+{
+	int n, arr[100], num;	//Declarartion of required variables
+	cout << "Enter the number of elements of the array: ";
+	cin >> n;	//Getting the number of elements of the array from the user
+	cout << endl << "Enter the elements of the array to be sorted: " << "\n";
```
Comment (HarshCasper) [inline docs/sorting-algorithms/bubble-sort.md:70]: Don't use comments beside the line of code. Use them above it.

File: docs/sorting-algorithms/bubble-sort.md [hunk 8]
```diff
@@ -3,4 +3,162 @@ title: Bubble Sort
...
+Third Pass:
+( 3 2 1 4 5 ) –> ( 2 3 1 4 5 ), Swap since 3 > 2
+( 2 3 1 4 5 ) –>  ( 2 1 3 4 5 ), Swap since 3 > 1
+( 2 1 3 4 5 ) –>  ( 2 1 3 4 5 ), No swap
+
+Fourth Pass:
+( 2 1 3 4 5 ) –> ( 1 2 3 4 5 ), Swap since 2 > 1
+( 2 3 1 4 5 ) –>  ( 2 1 3 4 5 ), No swap
+
+```
```
Comment (HarshCasper) [inline docs/sorting-algorithms/bubble-sort.md:37]: A diagram would be better here

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by shonali2600** [comment]
@HarshCasper made all the required changes, kindly review it.

File: docs/sorting-algorithms/bubble-sort.md [hunk 9]
```diff
@@ -3,4 +3,153 @@ title: Bubble Sort
...
+
+Let's have an array arr[] where we have the following elements: 5 4 3 2 1
+
+![Imgur](https://i.imgur.com/mrC2YZn.png)
+
+![Imgur](https://i.imgur.com/snYKCMG.png)
+
+![Imgur](https://i.imgur.com/blbuUB2.png)
+
+![Imgur](https://i.imgur.com/XLxdlis.png)
```
Comment (HarshCasper) [inline docs/sorting-algorithms/bubble-sort.md:20]: Did you make these diagrams?

File: docs/sorting-algorithms/bubble-sort.md [hunk 10]
```diff
@@ -3,4 +3,153 @@ title: Bubble Sort
...
 
-This page would serve our needs for documentation of Bubble Sort.
\ No newline at end of file
+## Introduction
+
+The bubble sort is a simple sorting algorithm which compares two adjacent elements and swaps them if they are not in the correct order.
+
+## Explanation
+
+Let's have an array arr[] where we have the following elements: 5 4 3 2 1
```
Comment (HarshCasper) [inline docs/sorting-algorithms/bubble-sort.md:12]: Keep the numbers in backticks

File: docs/sorting-algorithms/bubble-sort.md [hunk 11]
```diff
@@ -3,4 +3,153 @@ title: Bubble Sort
...
+- Time  complexity - O(n^2)
+- Space complexity - O(1) 
+
+```
+
+## Credits
+
+- [Rohan Babbar](https://github.com/rohanbabbar04) contributed the [C++ implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/sort/Bubble_Sort.cpp)
+- [Saloni Swagata](https://github.com/SaloniSwagata) contributed the [Python implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/sort/Bubble_Sort.py)
+
```
Comment (HarshCasper) [inline docs/sorting-algorithms/bubble-sort.md:155]: Remove the extra line

File: docs/sorting-algorithms/bubble-sort.md [hunk 12]
```diff
@@ -3,4 +3,153 @@ title: Bubble Sort
...
+1 2 3 5 6 8 9 10
+```
+
+## Complexity Analysis
+
+```
+- Time  complexity - O(n^2)
+- Space complexity - O(1) 
+
+```
```
Comment (HarshCasper) [inline docs/sorting-algorithms/bubble-sort.md:149]: No need to keep them under backticks. Just keep the value under backticks

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Can you please rebase against the `main` branch as well?

File: docs/sorting-algorithms/bubble-sort.md (hunk 9 above)
Comment (shonali2600) [reply docs/sorting-algorithms/bubble-sort.md:20]: Yes I made these diagrams on PowerPoint and then uploaded them to Imgur

**Comment by shonali2600** [comment]
@HarshCasper, I wrote `git rebase main` on git bash and it is displaying `Current branch 206 is up to date `. I don't know about rebase, please let me know if I tried to rebase it the right way

**Comment by shonali2600** [comment]
@HarshCasper kindly review this PR so that I can make changes if needed

**Review by ankitaggarwal23 — COMMENTED** [review]
Closing this due to inactivity since last 21 days. Thank you for your valuable time.😊

**Comment by shonali2600** [comment]
@ankitaggarwal23 It's the inactivity of PA , I asked several times, also I messaged him in discord, but got no reply from him.
I have put my efforts into this PR, made the required changes what @HarshCasper asked me to do.
So kindly review this PR, or ask @HarshCasper to review.

**Comment by salil-naik** [comment]
Any issues here @HarshCasper?

**Comment by ankitaggarwal23** [comment]
Actually, @HarshCasper already requested changes but no changes have been done.

**Comment by ankitaggarwal23** [comment]
We already informed in our channel try to tag at least 3 mentors so that review will be done faster but you are only tagging @HarshCasper then how others will come to know.

**Comment by shonali2600** [comment]
I made all the required changes what he asked me to, and after rebasing don't know why those changes are not appearing here.
I don't know what rebasing do.
I'll again push all the changes

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by ankitaggarwal23** [comment]
Thank you for your contribution.😊
@all-contributors please add @shonali2600 for his code

**Pull request merged**